miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultRunService.java

```diff
@@ -605,6 +609,57 @@ public boolean processNotification(Run source, int laneCount, String containerSe
     return isNew;
   }
 
+  private boolean updateMetricsFromNotification(Run source, Run target) {
+    if (source.getMetrics() != null && source.getMetrics().equals(target.getMetrics())) return false;
+    if (source.getMetrics() == null && target.getMetrics() != null) {
+      return false;
+    }
+    if (source.getMetrics() != null && target.getMetrics() == null) {
+      target.setMetrics(source.getMetrics());
+      return true;
+    }
+
+    ObjectMapper mapper = new ObjectMapper();
+    ArrayNode sourceMetrics;
+    try {
+      sourceMetrics = mapper.readValue(source.getMetrics(), ArrayNode.class);
+    } catch (IOException e) {
+      log.debug("Impossible junk metrics were passed in for run " + target.getId(), e);
```

Comment:
Should probably be concerning. IOException may be thrown for other reasons too. Log at error level